### PR TITLE
Add PowerShell completions

### PR DESCRIPTION
**Pull request by sequencerr**
### What does this PR do?

Closes: https://github.com/oven-sh/bun/issues/8939

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

**Comment by paperclover** [comment]
have not read through in depth but this looks legendary

**Comment by sequencerr** [comment]
<3 thanks. there are still some todos, link unlink completions, replace `get bins` with `bun.exe get completes [a-z]` (most are crashing on win). and zig part is pseudocoded (need help with this)

File: completions/bun.ps1 [hunk 1]
```diff
@@ -0,0 +1,1147 @@
...
+}
+
+$_bun_register_completer = {
+  param($wordToComplete, $commandAst, $cursorPosition)
+
+  $param = & $_bun_current_param $commandAst $cursorPosition
+  ((& $_bun_get_completes $commandAst $param) | Where-Object { $_.CompletionText -like "$wordToComplete*" })
+}
+
+Microsoft.PowerShell.Core\Register-ArgumentCompleter -CommandName bun -Native -ScriptBlock $_bun_register_completer
```
Comment (sequencerr) [inline completions/bun.ps1:1147]: It's possible to define `bun, bun.exe` at once. Do we need this?

**Review by paperclover — CHANGES_REQUESTED** [review]
I think the approach for this should deviate from what was taken for bash, zsh, and fish. See [this comment](https://github.com/oven-sh/bun/issues/8939#issuecomment-2033044246) in the powershell completions issue.

Instead of a 1.1k line addition to the user's powershell startup profile, this would be better done as a part of `bun getcompletes` and uses what bcdev-com suggests in [this comment](https://github.com/oven-sh/bun/issues/8939#issuecomment-2040644434)

**Pull request closed without merging**